### DISCUSSION TITLE
Fractional delay filter

### DISCUSSION THREAD

**Discussion by zhiha** [NVlabs/sionna#130] [Q&A]
Is there currently a library function in Sionna that can do the fractional delay of the signal?

I know some method like FFT can simply do that, but It would be even better if Sionna already supports the Farrow Method.

^_^

**Comment by jhoydis** [comment]
Hi,

There is no function that does this out-of-the-box. Wouldn't it be sufficient to define a [custom filter](https://nvlabs.github.io/sionna/api/signal.html#customfilter) corresponding to a time-shifted (truncated) sinc filter?

**Comment by nbecker** [comment]
fft is one approach.  Another relatively low-complexity approach is
Lagrange interpolator.

On Mon, May 15, 2023 at 2:51 PM Jakob Hoydis ***@***.***>
wrote:

> Hi,
>
> There is no function that does this out-of-the-box. Wouldn't it be
> sufficient to define a custom filter
> <https://nvlabs.github.io/sionna/api/signal.html#customfilter>
> corresponding to a time-shifted (truncated) sinc filter?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NVlabs/sionna/discussions/130#discussioncomment-5908628>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAA4VUCUBTLLCLRJLH36R6DXGJ3KTANCNFSM6AAAAAAX5PWOCE>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
*Those who don't understand recursion are doomed to repeat it*

**Reply by jhoydis** [reply]
The FFT-based solution could make for a nice contribution. We would be happy to review a pull request.
